### PR TITLE
JobExecution improvements

### PR DESCRIPTION
**Pull request by ivanvc**
1. Answer with a 404 when the JobTemplate doesn't exist
2. Add JobTemplate labels to JobExecution

### CODE REVIEW

File: pkg/http/execute_job_handler.go [hunk 1]
```diff
@@ -74,11 +83,18 @@ func createJobExecution(name, ns string, body io.ReadCloser) *v1alpha1.JobExecut
...
 		Spec: v1alpha1.JobExecutionSpec{
 			JobTemplateName: name,
 			Payload:         b.String(),
 		},
 	}
 }
+
+func (e *executeJobHandler) getJobTemplate(namespace, name string, ctx context.Context) (*v1alpha1.JobTemplate, error) {
+	jt := new(v1alpha1.JobTemplate)
+	err := e.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, jt)
```
Comment (tomcruz17) [inline pkg/http/execute_job_handler.go:98]: More of a learning question for me. I'm lost how `e` has a `Get` method. Is it some kind of inheritance from the *Server type in the struct? I know go doesn't have inheritance, so this is a mystery to me.

**Review by tomcruz17 — APPROVED** [review]
LGTM! Just have a question.

File: pkg/http/execute_job_handler.go (hunk 1 above)
Comment (ivanvc) [reply pkg/http/execute_job_handler.go:98]: It's called embedding structures/embedding anonymous structures (or you can even find it if you search inheritance in Go, although is more like composition): https://golangbyexample.com/embedding-interfaces-go/

Comment (tomcruz17) [reply pkg/http/execute_job_handler.go:98]: Thank you! Now I understand this `Get` method comes from a nested k8s `client.Client`. This embedding of anonymous structures looks powerful but can be confusing at times i.e. it's not clear which embedded struct/interface contains the method being invoked. And what of clashes in embedded structs/interface method names? I guess those should be compile time errors.

Just thinking out loud, and appreciating the learning!

Comment (ivanvc) [reply pkg/http/execute_job_handler.go:98]: Yes, I think you'll get a compilation error. However, you can still call `client.Client`'s `Get` function by doing `e.Client.Get(...)`

**Pull request merged**